noxfile.py

```diff
@@ -124,3 +131,11 @@ def run_python_integration_tests_with_db(session: Session):
         str(integration_test_directory),
         *session.posargs,
     )
+
+
+@nox.session(python=False)
+def update_user_guide(session: Session):
+    """
+    This task updates the examples in the user guide.
+    """
+    update_examples(Path("doc/user_guide/user_guide.md"))
```

Comment:
should this be plugged in to the prepare-release hooks to?

Reply:
Currently, it's separate.
There is a pre-commit hook and a github workflow verifying the user guide to be up-to-date. 
So moving it into the prepare-release is not required in my eyes. 
But in general, we could evaluate this idea - it sounds neither bad nor hard to accomplish to me.